File: Java/ArmstrongNumber.java
import java.util.Scanner;
import java.math.*;

public class ArmstrongNumber {
    public static void main(String args[]) {
        System.out.println("Enter your number to check if it is a Armstrong Number or not: ");
        Scanner sc = new Scanner(System.in);
        int x = sc.nextInt();
        int test = x, sum = 0, i = 0;
        while (test > 0) {
            test /= 10;
            i++;

        }
        test = x;
        while (test > 0) {
            int rem = test % 10;
            sum += Math.pow(rem, i);
            test /= 10;

        }
        if (x == sum) {
            System.out.println(x + " is a armstrong number!!");
        } else {
            System.out.println(x + " is not a armstrong number!!");
        }

    }
}

File: C++/Armstrongnumber.cpp
#include <iostream>
using namespace std;
int main()
{
    int newN;
    cout<<"Enter A number to check : ";
    cin >> newN;
    int n = newN;
    int temp;
    // int sum=0;
    int arm = 0;
    while (n)
    {
        temp = n % 10;
        n /= 10;
        // sum+=temp;
        cout << temp << " ";
        cout << endl;
        arm += temp * temp * temp;
        // cout<<"Sum"<<sum<<endl;
    }
    cout << endl;
    // cout<<arm;
    // cout<<newN;
    if (arm == newN)
    {
        cout << newN << " is a Armstrong number";
    }
    else
    {
        cout << newN << " is not a Armstrong number";
    }

    return 0;
}
